target/linux/octeon/base-files/lib/preinit/01_sysinfo

```diff
@@ -1,31 +1,69 @@
+do_get_shield_mode() {
+        # Figure out what mode the Shield is in
+        # We need to determine what Mode the Shield is in - Time to query the GPIOs
+        GPIO=99
+        GPIO16=99
+        GPIO17=99
+
+        # GPIO16 decides Recovery (formally Gateway) Mode.
+        GPIO=16
+        echo ${GPIO} > /sys/class/gpio/export
+        echo in > /sys/class/gpio/gpio$GPIO/direction
+        GPIO16=$(cat /sys/class/gpio/gpio${GPIO}/value)
+        echo $GPIO > /sys/class/gpio/unexport
+
+        # GPIO17 decides between Bridge and Router Mode
+        GPIO=17
+        echo $GPIO > /sys/class/gpio/export
+        echo in > /sys/class/gpio/gpio$GPIO/direction
+        GPIO17=$(cat /sys/class/gpio/gpio${GPIO}/value)
+        echo $GPIO > /sys/class/gpio/unexport
+
+        if [ $GPIO16 -eq 0 ]; then
+           # Recovery Mode
+           echo "itusgateway"
+        else
+           if [ $GPIO17 -eq 0 ]; then
+              # Bridge Mode
+                 echo "itusbridge"
+           else
+              # Router Mode
+                 echo "itusrouter"
+           fi
+        fi
+}
+
 do_sysinfo_octeon() {
-	local machine
-	local name
+        local machine
+        local name
 
-	machine=$(grep "^system type" /proc/cpuinfo | sed "s/system type.*: \(.*\)/\1/g")
+        machine=$(grep "^system type" /proc/cpuinfo | sed "s/system type.*: \(.*\)/\1/g")
+        smode=$(do_get_shield_mode)
 
-	case "$machine" in
-	"UBNT_E100"*)
-		name="erlite"
-		;;
+        case "$machine" in
+        "UBNT_E100"*)
+                name="erlite"
+                ;;
 
-	"UBNT_E200"*)
-		name="er"
-		;;
+        "UBNT_E200"*)
+                name="er"
+                ;;
 
-	"UBNT_E220"*)
-		name="erpro"
-		;;
+        "UBNT_E220"*)
+                name="erpro"
+                ;;
 
-	*)
-		name="generic"
-		;;
-	esac
+        "ITUS_SHIELD"*)
+                name=$smode
+                ;;
+        *)
+                name="generic"
+                ;;
+        esac
 
-	[ -e "/tmp/sysinfo/" ] || mkdir -p "/tmp/sysinfo/"
+        [ -e "/tmp/sysinfo/" ] || mkdir -p "/tmp/sysinfo/"
 
-	echo "$name" > /tmp/sysinfo/board_name
-	echo "$machine" > /tmp/sysinfo/model
+        echo "$name" > /tmp/sysinfo/board_name
+        echo "$machine" > /tmp/sysinfo/model
```

Comment:
Changing these indents is unrelated and shouldn't be done anyway, as we prefer tabs here (traditionally).

Reply:
Will correct